internal/translayer/dbuser/translation.go

```diff
@@ -0,0 +1,183 @@
+package dbuser
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/timeutil"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type User struct {
+	akov2.AtlasDatabaseUserSpec
+	Password  string
+	ProjectID string
+}
+
+func NewUser(spec akov2.AtlasDatabaseUserSpec, projectID, password string) *User {
```

Comment:
nit: avoid shallow-copies, this produces unnecessary memory pressure and doesn't help with nested maps and slices. I vote for simply:
```suggestion
func NewUser(spec *akov2.AtlasDatabaseUserSpec, projectID, password string) *User {
```
and a godoc whether `spec` is being mutated signalling the caller potentially has to `spec.DeepCopy` first. Note that DeepCopying should always be done on the consumer side, never here. This is the convention used in the Kubernetes source code base and other operators as well.

Reply:
Will fix and document

Reply:
coming next